rotatescreen/display_linux.py

```diff
@@ -0,0 +1,110 @@
+from Xlib import display
+from Xlib.ext import randr 
+from typing import Dict, List, Tuple
+
+# TODO: Is having these at the global level ok?
+d = display.Display()
+# Create an X display and get the root window
+root = d.screen().root
+# Get the resources for the root window
+res = root.xrandr_get_screen_resources()
+
+
+class Display:
+
+    def __init__(self, output_id, crtc_id):
+        self.output_id = output_id
+        self.crtc_id = crtc_id
+
+    def __repr__(self):
+        return f"<'{self.device_description[0]}' object>"
+
+    def rotate_to(self, degrees: int) -> None:
+        if degrees == 90:
+            rotation_val = randr.Rotate_90
+        elif degrees == 180:
+            rotation_val = randr.Rotate_180
+        elif degrees == 270:
+            rotation_val = randr.Rotate_270
+        elif degrees == 0:
+            rotation_val = randr.Rotate_0
+        else:
+            raise ValueError("Display can only be rotated to 0, 90, 180, or 270 degrees.")
+        
+        # Get the CRTC's current mode and position
+        crtc = d.xrandr_get_crtc_info(self.crtc_id, res.config_timestamp)
+        x, y, mode = crtc.x, crtc.y, crtc.mode
+        
+        # Rotate the CRTC 90 degrees clockwise
+        d.xrandr_set_crtc_config(    
+            crtc=self.crtc_id,
+            config_timestamp=res.config_timestamp,
+            x=x,
+            y=y,
+            mode=mode,
+            rotation=rotation_val,
+            outputs=crtc.outputs
+        )
+
+    def set_landscape(self) -> None:
+        self.rotate_to(0)
+
+    def set_landscape_flipped(self) -> None:
+        self.rotate_to(180)
+
+    def set_portrait(self) -> None:
```

Comment:
fwiw, arandr and xrandr both call this `right` and `left` instead of `portrait` and `portrait_flipped`, i'm happy to add a seperate function in both this and the windows one for people who like that naming scheme better, as you mentioned wanting to keep the APIs the same for both.

Reply:
I've had a think about this one; 

There's also an argument to commit to the breaking change and just to rename them if `left`, `right`, `normal`, `inverted` are considered clearer - which I _could_ understand however it also seems quite subjective and I'm reluctant to break the interface.

I've gone with method aliases for now but I'm not sure if I'll keep this. My main objection being it doesn't feel like a very pythonic solution - the whole "There should be one-- and preferably only one --obvious way to do it." ([PEP-0020](https://peps.python.org/pep-0020/))... but perhaps this will suffice for now.

Reply:
method aliases sound like a decent way of doing it, i think that committing to one style only isnt great. I think that it is preferable to make something clear that users are used to than to break a 'pythonic' way of doing things